frontends/web/src/routes/device/bitbox02/settings.tsx

```diff
@@ -29,135 +29,106 @@ import { GotoStartupSettings } from './gotostartupsettings';
 import { Reset } from './reset';
 import { SetDeviceName } from './setdevicename';
 import { ShowMnemonic } from './showmnemonic';
-import { UpgradeButton, VersionInfo } from './upgradebutton';
+import { UpgradeButton } from './upgradebutton';
 import { alertUser } from '../../../components/alert/Alert';
 
-interface SettingsProps {
+interface Props {
     deviceID: string;
 }
 
-interface State {
-    versionInfo?: VersionInfo;
-    deviceInfo?: DeviceInfo;
-}
-
-type Props = SettingsProps & TranslateProps;
-
-class Settings extends Component<Props, State> {
-    public readonly state: State = {};
-
-    private apiPrefix = () => {
-        return 'devices/bitbox02/' + this.props.deviceID;
-    }
+export const Settings: FunctionComponent<Props> = ({ deviceID }) => {
+    const { t } = useTranslation();
+    const [deviceInfo, setDeviceInfo] = useState<DeviceInfo>();
+    const getInfo = () => {
+        getDeviceInfo(deviceID).then(setDeviceInfo).catch(error => {
+            console.error(error);
+            alertUser(t('genericError'));
+        });
+    };
+    useEffect(getInfo, [deviceID, t]);
 
-    private getInfo = () => {
-        getDeviceInfo(this.props.deviceID)
-            .then(deviceInfo => this.setState({ deviceInfo }))
-            .catch(error => {
-                console.error(error);
-                alertUser(this.props.t('genericError'));
-            });
-    }
+    const versionInfo: VersionInfo | undefined = useLoad(() => getVersion(deviceID), [deviceID]);
```

Comment:
`VersionInfo | undefined` is already the return type from useLoad+getVersion, no need to type again.

![Screen Shot 2022-04-26 at 4 27 17 PM](https://user-images.githubusercontent.com/546900/165323065-a346b415-6936-4cfb-a152-17f8fe8ee014.png)

Reply:
removed